front/pages/[user]/a/[sId]/execute/index.jsx

```diff
@@ -0,0 +1,392 @@
+        <input
+          className={classNames(
+            "px-1 py-1 font-normal text-[13px] font-mono bg-slate-300 border-0 outline-none focus:outline-none w-full",
+            "border-white ring-0 focus:ring-0 focus:border-white"
+          )}
+          readOnly={true}
+          value={inputName}
+        />
+      </div>
+      <div
+        className={classNames(
+          "col-span-7 inline-grid space-y-0 resize-none text-[13px] font-mono px-0 py-0 border bg-slate-100",
+          "border-slate-100"
+        )}
+      >
+        <TextareaAutosize
+          minRows={1}
+          className={classNames(
+            "w-full resize-none font-normal text-[13px] font-mono px-1 py-0 bg-transparent border-0 ring-0 focus:ring-0",
+            "text-gray-700"
+          )}
+          value={inputValue || ""}
+          onChange={(e) => onChange(e.target.value)}
+        />
+      </div>
+    </div>
+  );
+}
+
+export default function ExecuteView({
+  app,
+  user,
+  ga_tracking_id,
+  inputDataset,
+  savedSpecification,
+  config,
+}) {
+  const { data: session } = useSession();
+
+  const [inputDatasetKeys, _setInputDatasetKeys] = useState(
+    inputDataset ? checkDatasetData(inputDataset.dataset.data, false) : []
+  );
+
+  const [inputData, setInputData] = useState({});
+  const isInputDataValid = () =>
+    inputDatasetKeys.every((k) => (inputData[k] || "").length > 0);
+
+  const [isRunning, setIsRunning] = useState(false);
+  const [isRunComplete, setIsRunComplete] = useState(false);
+
+  const [executionLogs, setExecutionLogs] = useState({
+    blockOrder: [],
+    lastEventByBlockName: {},
+    outputByBlockName: {},
+  });
+
+  const [outputExpandedByBlockName, setOutputExpandedByBlockName] = useState(
+    {}
+  );
+
+  const {
+    run: savedRun,
+    _isRunLoading,
+    _isRunError,
+  } = useSavedRunStatus(user, app, (data) => {
+    if (data && data.run) {
+      switch (data?.run.status.run) {
+        case "running":
+          return 100;
+        default:
+          return 0;
+      }
+    }
+    return 0;
+  });
+
+  const expandLastBlockOutput = () => {
+    const lastBlockName =
+      executionLogs.blockOrder[executionLogs.blockOrder.length - 1];
+    setOutputExpandedByBlockName({
+      ...outputExpandedByBlockName,
+      [lastBlockName]: true,
+    });
+  };
+
+  useEffect(() => {
+    if (isRunComplete) {
+      expandLastBlockOutput();
+    }
+  }, [isRunComplete]);
+
+  const handleValueChange = (k, value) => {
+    const newInputData = { [k]: value };
+    setInputData({ ...inputData, ...newInputData });
+  };
+
+  const handleRun = () => {
+    setExecutionLogs({
+      blockOrder: [],
+      lastEventByBlockName: {},
+      outputByBlockName: {},
+    });
+    setIsRunning(true);
+    setIsRunComplete(false);
+    setOutputExpandedByBlockName({});
+
+    setTimeout(async () => {
+      const specificationHash = savedRun?.app_hash;
+
+      const requestBody = {
+        specification: specificationHash
```

Comment:
I think we should only support savedRun.app_hash and nothing else.

We can show the hash here and if no run were made, we can disable the run button

Reply:
Do you mean display the hash ?